pipeline/vars/bm_send_test_reports.groovy

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env groovy
+/*
+    Groovy script to publish the report generated by baremetal specific pipeline
+    Accepts test results <MAP> as parameter,
+    e.g :results = ["StageName1":[suite1: [Result: "Pass", Logs:"path to log"],
+                                  suite2: [Result: "PASS", Logs:"path to log"],
+                                  suite6: [Result: "PASS", Logs:"path to log"]],
+                    "StageName2":[suite3: [Result: "Pass", Logs:"path to log"],
+                                  suite4: [Result: "Fail", Logs:"path to log"],
+                                  suite5: [Result: "Fail", Logs:"path to log"]],
+                    "StageName3":[suite13: [Result: "Pass", Logs:"path to log"]]]
+*/
+
+def sendEMail(def test_results) {
+    /*
+        Send Email notification mentioning the stages, corresponding
+        suites and their execution status
+    */
+    def body = new File("pipeline/vars/emailable-report.html").text
+    def gchat_result = ""
+    body += "<h2><u>Test Artifacts</h2></u><table><tr><td>COMPOSE_URL </td><td>${env.composeUrl}</td></tr><td>COMPOSE_ID</td><td> ${env.composeId}</td></tr>"
+    body += "<tr><td> REPOSITORY </td><td>${env.repository}</td></tr>"
+
+    body += "</table>"
+    body += "<body><u><h3>Test Summary</h3></u><br />"
+    body += "<p>Logs are available at ${env.BUILD_URL}</p><br />"
+
+    body += "<table>"
+    heading = "<tr><th>Stage</th><th>Suite</th><th>Result</th><th>Logs</th></tr>"
+    body += heading
+    def status = "PASSED"
+    for ( stage in test_results ) {
+        body += "<tr><td rowspan = \"${stage.value.size()}\">${stage.key}</td>"
+        gchat_result+= "\nSTAGE : ${stage.key}"
+        for ( suites in stage.value ) {
+            res = suites.value["Result"]
+            body += "<td>${suites.key}</td>"
+            body += "<td>${suites.value["Result"]}</td>"
+            body += "<td>${suites.value["Logs"]}</td>"
+            body += "</tr>"
+            gchat_result += "\n\tSuite Name : ${suites.key}"
+            gchat_result += "\n\t\tResult : ${suites.value["Result"]}"
+            gchat_result += "\n\t\tLogs : ${suites.value["Logs"]}"
+            if (suites.value["Result"].toLowerCase() != "pass"){
+                status = "FAILED"
+            }
+        }
+        gchat_result += "\n\n"
+    }
+    body +="</table> </body> </html>"
+
+    def rh_ceph_version = env.rhcephVersion.substring(0,3)
+    subject = "[UPI Pipeline] [${rh_ceph_version}]Test execution status of ${env.JOB_NAME} - ${env.BUILD_NUMBER} : ${status}"
```

Comment:
sample output from sample jenkins job : 

[UPI Pipeline] Test execution status of test_job - 7 : PASS

Excluded rh_ceph_version from output as it was a sample job and that env variable was not present